### PR TITLE
Improve swift attribute formatting

### PR DESCRIPTION
**Pull request by mportiz08**
Bug/issue #, if applicable: 108412440

## Summary

Updates the formatting logic for Swift declarations to break each [attribute][1] onto its own line.

Example:
```swift
// before
@discardableResult @objc(baz) func foobarbaz() -> Int

// after
@discardableResult
@objc(baz)
func foobarbaz() -> Int
```

\cc @ethan-kusters 

## Testing

Steps:
1. Configure [swift-docc-plugin][2] for a new or existing swift package that has examples of Swift symbols with attributes
2. Build this branch and note the location of `dist`
3. Run `DOCC_HTML_DIR=/path/to/dist swift package --disable-sandbox preview-documentation --target [target]` to preview it and verify that the symbols now have each attribute on its own line.

Here is an example package (target `AttributeExamples`) for convenience:
[AttributeExamples.zip](https://github.com/apple/swift-docc-render/files/11338015/AttributeExamples.zip)

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [X] Added tests
- [X] Ran `npm test`, and it succeeded
- [X] Updated documentation if necessary

[1]: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/attributes
[2]: https://github.com/apple/swift-docc-plugin

### CODE REVIEW

**Comment by ethan-kusters** [comment]
> ```swift
> // before
> @discardableResult @objc(baz) func foobarbaz() -> Int
> 
> // after
> @discardableResult
> @objc(baz)
> func foobarbaz() -> Int
> ```

Did you consider putting all attributes on a single line instead of an additional line per attribute? Most attributes are pretty short so I think that _may_ be a better balance here. I don't have a very strong opinion either way though – we can always make adjustments later after we see how things look in-practice.

```swift
// before
@discardableResult @objc(baz) func foobarbaz() -> Int

// after
@discardableResult @objc(baz)
func foobarbaz() -> Int
```

**Comment by ethan-kusters** [comment]
> I didn't give it much consideration. From looking at code examples, I didn't see many scenarios with more than one attribute, and it seemed that the common pattern was splitting the attributes onto their own lines or having everything inline.
> 
> It would be a more involved implementation, but I could look into that if we wanted to pursue the design direction of having all attributes on one line.

I think you're right that the vast-majority of current use-cases involve a single attribute. If this is the simpler implementation I think it makes sense to start with this and we can adjust in the future if there's a clear need.

File: src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue [hunk 1]
```diff
@@ -127,6 +127,18 @@ export default {
           }
         }
 
+        // check if this is a text token following an attribute token
+        // so we can insert a newline here and split each attribute onto its
+        // own line
+        //
+        // we want to avoid doing this when the attribute is encountered
+        // in a param clause for attributes like `@escaping`
```
Comment (hqhhuang) [inline src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue:135]: Looks great! Can we add a test for this scenario?

Comment (mportiz08) [reply src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue:135]: 0cc2a2476858c45cd0c376444cc12c8edbdeaf6c

**Review by hqhhuang — APPROVED** [review]
LGTM!

**Comment by mportiz08** [comment]
@swift-ci test

**Pull request merged**